chisel/src/main.rs

```diff
@@ -1,30 +1,255 @@
     }
+}
+
+fn execute_module(context: &ModuleContext, module: &Module) -> bool {
+    let (conf_name, conf_preset) = context.fields();
+    let preset = conf_preset
+        .clone()
+        .unwrap_or(String::from("ewasm"))
+        .to_string();
+
+    let name = conf_name.as_str();
+    match name {
+        "verifyexports" => {
+            if let Ok(chisel) = VerifyExports::with_preset(&preset) {
+                chisel.validate(module).unwrap_or(false)
+            } else {
+                false
+            }
+        },
+        "verifyimports" => {
+            if let Ok(chisel) = VerifyExports::with_preset(&preset) {
+                chisel.validate(module).unwrap_or(false)
+            } else {
+                false
+            }
+        },
+        "checkstartfunc" => {
+            //NOTE: checkstartfunc takes a bool for configuration. false by default for now.
+            let chisel = CheckStartFunc::new(false);
+            let ret = chisel.validate(module).unwrap_or(false);
+            ret
+        }, /*
+        "deployer" => 
+        "trimexports"
+        "remapimports"
+        */
+        _ => false,
+    }
+}
 
-    let code = std::fs::read(&args[1]).expect("Failed to open and read file");
+fn chisel_execute(context: &ChiselContext) -> Result<bool, &'static str> {
+    if let Ok(buffer) = read(context.file()) {
+        if let Ok(module) = deserialize_buffer::<Module>(&buffer) {
+            let chisel_results = context
+                .get_modules()
+                .iter()
+                .map(|ctx| execute_module(ctx, &module))
+                .fold(true, |b, e| e & b);
+            Ok(chisel_results)
+        } else {
+            Err(ERR_DESERIALIZE_MODULE)
+        }
+    } else {
+        Err(ERR_FAILED_OPEN_BINARY)
+    }
+}
 
-    let mut module = parity_wasm::deserialize_buffer(&code).expect("Failed to load module");
+fn chisel_subcommand_run(args: &ArgMatches) -> i32 {
+    let config_path = args.value_of("CONFIG").unwrap_or(DEFAULT_CONFIG_PATH);
 
-    let trimexports = trimexports::TrimExports::with_preset("ewasm").unwrap();
-    trimexports
-        .translate(&mut module)
-        .expect("Failed to trim exports");
+    if let Ok(conf) = read_to_string(config_path) {
+        match yaml_configure(conf) {
+            Ok(ctx) => match chisel_execute(&ctx) {
+                Ok(result) => if result {
+                    return 0;
+                } else {
+                    return 1;
+                },
+                Err(msg) => err_exit(msg),
+            },
+            Err(msg) => err_exit(msg),
+        };
+    } else {
+        err_exit(ERR_FAILED_OPEN_CONFIG);
+    }
+}
 
-    let remapimports = remapimports::RemapImports::with_preset("ewasm").unwrap();
-    remapimports
-        .translate(&mut module)
-        .expect("Failed to remap imports");
+pub fn main() {
+    let cli_matches = App::new("chisel")
+        .version(crate_version!())
+        .author(crate_authors!())
```

Comment:
This generates some weird output:
```
Alex Beregszaszi <alex@rtfs.hu>:Jake Lang <jak3lang@gmail.com
```

Can we just remove this line?

Reply:
done.